docs/root/intro/arch_overview/matching/matching_api.rst

```diff
@@ -0,0 +1,112 @@
+.. _arch_overview_matching_api:
+
+Matching API
+============
+
+Envoy makes use of a matching API to allow the various subsystems to express actions that should be performed based on incoming data.
+
+The matching API is designed as a tree structure to allow for sublinear matching algorithms, and make heavy use of extension points to make it
```

Comment:
Nit: "makes heave use". I'd also maybe say "for performance better than linear list matching as seen in Envoy's HTTP routing" or the like to clearly provide the bogeyman that is at stake.